sktime/classification/shapelet_based/_rocket_classifier.py

```diff
@@ -17,6 +18,14 @@
 from sktime.utils.validation.panel import check_X_y
 
 
+def _fit_pipeline(X, y, num_kernels, random_state):
```

Comment:
I suggest to construct the instance once outside the function in the main `fit` method, and then have this function accept a cloned instance and X and y, rather than passing hyper-parameters and constructing it inside the function (for cloning you can use `from sklearn.base import clone`)

```python
def _fit_estimator(estimator, X, y):
    return estimator.fit(X, y)
```

Reply:
If I understand this correctly, we also want to make sure that ensemble members use different random states (though fixed if passed), to avoid having an ensemble in which all members are the same, I remember that scikit-learn has some helper function for this but don't remember where exactly

Reply:
Regarding `random_state`, absolutely. I've been thinking about it but didn't find how exactly this is handled for other ensembles throughout the project. E.g. here https://github.com/alan-turing-institute/sktime/blob/30cb0fc3ae1001ed7936b82331627c734ea1698d/sktime/series_as_features/base/estimators/interval_based/_tsf.py#L162-L172 -- the same `random_state` is passed a parameter into all estimators. Let me check what does `sklearn` provide as an option

Reply:
Okay, I found how `sklearn` sets random state internally for ensembles. Working on update to the PR. I'm going to make `n_jobs=None` by default and have another branch where the processing happens exactly as it was done before (without any manipulations with `random_state`). Otherwise this update will break reproducibility of ensembles trained previously.